.sbtopts

```diff
@@ -0,0 +1,17 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+-J-Xmx4G
+-J-Xss4m
```

Comment:
@gemelen sorry for the very late comment but do you remember why we added this? The default memory has set in ` build/sbt-launch-lib.bash` (e.g., see https://github.com/apache/spark/commit/35bab339848cba7e518478cb87591bff6a95ff07). Were you using plain `sbt` instead of `build/sbt` in your local?

This file disables the memory option from the `build/sbt` script:

```bash
./build/sbt -mem 6144
```
```
.../jdk-11.0.3.jdk/Contents/Home as default JAVA_HOME.
Note, this will be overridden by -java-home if it is set.
Error occurred during initialization of VM
Initial heap size set to a larger value than the maximum heap size
```

because it adds these memory options at the last:

```bash
/.../bin/java -Xms6144m -Xmx6144m -XX:ReservedCodeCacheSize=256m -Xmx4G -Xss4m -jar build/sbt-launch-1.5.0.jar 1
```

and Java respects the rightmost memory configurations.

Reply:
Ohh, okay it sets the stack size here. I misread it. Okay but I will still move this to the script.

Reply:
@HyukjinKwon AFAIR stack size increase was introduced to overcome failures on some tasks (tests probably). Yep, definitely could be set anywhere it is more suitable

Reply:
Commit message says it was for Github Actions env in fact

Reply:
Thanks for explanation!